lib/meilisearch/index.rb

```diff
@@ -231,5 +232,16 @@ def accept_new_fields
     def update_accept_new_fields(accept_new_fields)
       http_post "/indexes/#{@uid}/settings/accept-new-fields", accept_new_fields
     end
+
+    def wait_for_pending_update(update_id, timeout_in_ms = 5000, interval_in_ms = 50)
+      Timeout.timeout(timeout_in_ms.to_f / 1000) do
```

Comment:
I would rather throw a custom MeiliSearch Exception so that the user knows it comes from MeiliSearch. You have to add an error in the `error.rb` file.
We do the same in JS and PHP for example:
- custom error: https://github.com/meilisearch/meilisearch-php/blob/master/src/Exceptions/TimeOutException.php
- the method: https://github.com/meilisearch/meilisearch-php/blob/13cc34d4ee19b7bbf674d8f9af57545c8d6338e9/src/Index.php#L94

(I realized we didn't do that for the python, we have to fix that, but it's normal because when we created this method, the error handler didn't exist ^^)

Reply:
I did implement a custom error, also with a custom message, let me know what do you think :)